### PR TITLE
feat: Implement with_menu_on_left_click on windows

### LINKED ISSUE #7719: [feat] Allow Windows system tray menu open on mouse/trackpad left click
### Describe the problem

On Windows, if I want to access (show) system tray menu, I need to click the right mouse/trackpad click (show menu) then left click to select menu item. This is ok by default if you don't need to access the menu too often but if have important functionality in the menu items, like dynamically generated menu items, it would be nice to have an option to allow users to show system tray menu and select items with the same mouse/trackpad button, `left click` for all actions, on all platforms. Currently on Mac and Linux you can use left click to open system tray menu and select items. Only Windows required to use the right click.

### Describe the solution you'd like

I would like to have an option to allow create a system tray menu on Windows with option to show the menu on left click, instead of standard action on right click. Maybe just add support for `SystemTray::with_menu_on_left_click` on Windows platform.

https://github.com/tauri-apps/tauri/commit/63011ca84e7a22c8c0d8bd1c1be6592140f93ff2#diff-6d57b6afece9589b880315885f1cce9b4e645c00428b503bb2720424b6bdea34

### Alternatives considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by htngr**
Fixes https://github.com/tauri-apps/tauri/issues/7719 (for tauri 2.0)

### CODE REVIEW

**Review by amrbashir — CHANGES_REQUESTED** [review]
Thank you, could you add a change file in `.changes` directory?

**Comment by htngr** [comment]
Done

File: .changes/with_menu_on_left_click_windows.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"tray-icon": "minor"
+---
+
+Implemented TrayIcon::with_menu_on_left_click on windows
```
Comment (amrbashir) [inline .changes/with_menu_on_left_click_windows.md]:
```suggestion
Implemented `TrayIcon::with_menu_on_left_click` on windows
```

**Review by amrbashir — APPROVED** [review]
Thank you

**Pull request merged**